src/lib/png/png.c

```diff
@@ -0,0 +1,1128 @@
+				}
+			} else if(code == 18) /* repeat "0" 11-138 times */ {
+				replength = 11; /* read in the bits that indicate repeat length */
+				if((*bp + 7) > in_bitlen) {
+					err = -1; /* error, bit pointer jumps past memory */
+					break;
+				}
+				replength += readBitsFromStream(bp, in, 7);
+
+				/* repeat this value in the next lengths */
+				for(n = 0; n < replength; n++) {
+					if(i >= HLIT + HDIST) {
+						err = -1; /* error: i is larger than the amount of codes */
+						break;
+					}
+					if(i < HLIT) {
+						bitlen_ll[i] = 0;
+					} else {
+						bitlen_d[i - HLIT] = 0;
+					}
+					i++;
+				}
+			} else {
+				if(code == -1) {
+					err = -1; /* error in decodesymbol fun */
+					break;
+				}
+			}
+		}
+		if(err) break;
+
+		if(bitlen_ll[256] == 0) {
+			err = -1; /* the length of the end code 256 must be larger than 0 */
+			break;
+		}
+
+		/* generate the code trees */
+		err = make_tree_first_step(ll_tree, bitlen_ll, NUMBER_OF_CODE_SYMBOLS, 15);
+		if(err) break;
+		err = make_tree_first_step(dist_tree, bitlen_d, NUMBER_OF_DISTANCE_SYMBOLS, 15);
+
+		break; /* end of while(!err) */
+	}
+	HuffmanTree_cleanup(&cl_tree);
+	return err;
+}
+
+/* get the dist_tree with fixed huffman */
+int build_FixeddTree(HuffmanTree *tree) {
+	int i, err = 0;
+	int bitlen[NUMBER_OF_DISTANCE_SYMBOLS];
+
+	for(i = 0; i != NUMBER_OF_DISTANCE_SYMBOLS; i++) {
+		bitlen[i] = 5;
+	}
+
+	err = make_tree_first_step(tree, bitlen, NUMBER_OF_DISTANCE_SYMBOLS, 15);
+	return err;
+}
+
+/* get the ll_tree with fixed huffman */
+int build_FixedllTree(HuffmanTree *tree) {
+	int i, err = 0;
+	int bitlen[NUMBER_OF_CODE_SYMBOLS];
+
+	for(i = 0; i <= 143; i++) {
+		bitlen[i] = 8;
+	}
+
+	for(i = 144; i <= 255; i++) {
+		bitlen[i] = 9;
+	}
+
+	for(i = 256; i <= 279; i++) {
+		bitlen[i] = 7;
+	}
+
+	for(i = 280; i <= 287; i++) {
+		bitlen[i] = 8;
+	}
+
+	err = make_tree_first_step(tree, bitlen, NUMBER_OF_CODE_SYMBOLS, 15);
+	return err;
+}
+
+/* get the trees with fixed huffman blocks */
+int build_FixedTrees(HuffmanTree *ll_tree, HuffmanTree *dist_tree) {
+	int err;
+	err = build_FixedllTree(ll_tree);
+	if (!err) {
+		err = build_FixeddTree(dist_tree);
+	}
+	return err;
+}
+
+int inflate_Huffman(unsigned char *out, unsigned char *in, int *bp,
+								int *pos, int in_length, int btype) {
+	int err = 0;
+	HuffmanTree ll_tree; /* literal and length codes */
+	HuffmanTree dist_tree; /* distance codes */
+	int in_bitlen = in_length * 8;
+	int l_code, length_base; /* ll code and length */
+	int dist_code, dist; /* distance code and distance */
+	int addit_bits_ll, addit_bits_d; /* extra bits for length and distance */
+	int first_pos, forw, back;
+
+	/* build Huffman trees */
+	HuffmanTree_init(&ll_tree);
+	HuffmanTree_init(&dist_tree);
+
+	if(btype == 1) {
+		err = build_FixedTrees(&ll_tree, &dist_tree);
+	} else if(btype == 2) {
+		err = build_DynamicTrees(&ll_tree, &dist_tree, in, bp, in_length);
+	}
+
+	while(!err) { /* decode symbols until error or end */
```

Comment:
You `break` the loop when `err` is being set, so this condition seems to be unneccessary

Reply:
fixed